libs/platform/ebpf_extension.c

```diff
@@ -417,3 +438,15 @@ ebpf_provider_unload(_Frees_ptr_opt_ ebpf_extension_provider_t* provider_context
     ebpf_free(provider_context);
     EBPF_RETURN_VOID();
 }
+
+bool
+ebpf_extension_lock_provider_data(_Inout_ ebpf_extension_client_t* client_context)
```

Comment:
Is it really locked for exclusive access? or is it just referenced?  The name in 445 implies the latter but the name in 443 implies the former.

Reply:
OK, maybe locked isn't the best name. It is prevented from unloading, hence locked in place, rather than exclusively accessed. 

Would reference / unreferenced be better?